### DISCUSSION TITLE
Git Bash push error

### DISCUSSION THREAD

**Discussion by cuneytda** [community/community#32039] [Other Feature Feedback, Questions, & Ideas]
Hello,
Command "git push -u origin master" in git bash
After that I get an error like this;

fatal: An error occurred while sending the request.
fatal: Request aborted: Failed to create SSL/TLS secure channel.
remote: Support for password authentication was removed on August 13, 2021.
remote: Please see https://docs.github.com/en/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls for information on currently recommended modes of authentication.
fatal: Authentication failed for 'https://github.com/username/repostory.git/'

What should I do?
Thank you.

**Comment by airtower-luna** [comment]
[Create a personal access token (PAT)](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) and input that instead of your password when Git asks. Your token needs to have the **repo** scope, and also **workflow** if you use Actions.

**Comment by cuneytda** [comment]
Firstly, thanks for your response. I did what you said "Create a personal access token (PAT) and input that instead of your password when Git asks.". But I am getting the same error..

**Comment by airtower-luna** [comment]
In that case there's probably some issue with how you input the token. That message appears whenever you enter something that *isn't* a valid token. Maybe an extra character (e.g. space) if you copy & paste the token, or a typo if you type it (which you probably should avoid)?

**Comment by cuneytda** [comment]
I only tick this box;
![github](https://user-images.githubusercontent.com/65169534/188497810-404dfa24-9fbb-4d80-87d1-6ddf66d02f10.png)

**Comment by airtower-luna** [comment]
That looks correct. What I mean is that there's probably some problem with how you get the generated token into the Git "password" prompt.

**Comment by cuneytda** [comment]
I enter the Token carefully but I get the same error..

**Comment by airtower-luna** [comment]
Did you try copying & pasting the token? There should be a copy button next to it when it's generated, so you can copy it without accidentally getting anything else.

**Comment by cuneytda** [comment]
Yes. I copying and pasting the token. I click copy button.

**Comment by cuneytda** [comment]
my git bash screen;
![github_2](https://user-images.githubusercontent.com/65169534/188500633-36d6b04c-021e-412a-b76e-b44a4c3e4ba2.png)

**Comment by airtower-luna** [comment]
Wait, I don't see any password prompt there. Is there some other kind of prompt, or does Git never ask you?

If it doesn't ask, you password may be cached somewhere ("credential manager" or similar), and you'd need to delete it first (or directly replace it with the token).

**Comment by cuneytda** [comment]
I deleted the github credential in "credential manager".

**Reply by cuneytda** [reply]
Where else can my password be stored?

**Reply by airtower-luna** [reply]
I'm not sure, but you can check the `credential.helper` entry in your Git config.

See also: https://git-scm.com/book/en/v2/Git-Tools-Credential-Storage

**Comment by cuneytda** [comment]
I'm pasting the personal access token in the wrong place. But now I am getting an error like this. I ran "git push" after "git branch -M main". Could it be because of this? What should I do?
![github_4](https://user-images.githubusercontent.com/65169534/188656838-2f33e595-407c-43b3-ab0f-c7d4c071a130.png)

**Comment by airtower-luna** [comment]
That means authorization works now. :slightly_smiling_face: 

That error message means that the branch in the remote repository (on GitHub) has commits that aren't on the local branch you're trying to push. How to solve that depends on whether you want to keep those commits.

If you do want to keep them, either _merge_ them into your local branch, or _rebase_ your local branch on the remote branch before pushing.

If you don't want to keep the remote commits you can force push to make Git overwrite the remote branch even if it means losing data. :warning:

**Reply by cuneytda** [reply]
Yes. Authorization is working. :)
But now I have another problem.
I think I will reinstall the remote repository..
Thank you very much for your contact. I'll write again if I can't handle it :)

**Reply by airtower-luna** [reply]
Glad I could help! Marking the answer would be appreciated, if you have another (different) problem it's best to make a new thread for a new question. :slightly_smiling_face:

**Reply by cuneytda** [reply]
👍
